pulp_rpm/app/downloaders.py

```diff
@@ -26,6 +30,18 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
 
+class NoUlnCredentialsError(Exception):
+    """
+    Exception to signal, that no valid ULN Credentials were given.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """
+        Exception to signal, that no valid ULN Credentials were given.
+        """
+        super().__init__("No valid ULN credentials given.", *args, **kwargs)
+
+
```

Comment:
there is a dedicated place for exceptions: `pulp_rpm/app/exceptions.py`, consider subclassing from `PulpException` to have a dedicated code and to be shown properly in the task status.

Reply:
Thanks. I moved it to the exception.py file. Done